reposerver/repository/repository.go

```diff
@@ -2512,7 +2512,7 @@ func (s *Service) ResolveRevision(ctx context.Context, q *apiclient.ResolveRevis
 	app := q.App
 	ambiguousRevision := q.AmbiguousRevision
 	var revision string
-	var source = app.Spec.GetSource()
+	var source = app.Spec.GetSourcePtrByIndex(int(q.SourceIndex))
```

Comment:
Not sure if it is possible that source can be nil, base on validations during app creation, but before this change, source was not reference and it always was not nil, after this change it may happens that you can get error here

Reply:
At any point, the application will have either at least 1 source in `spec.Sources` or `spec.Source` defined.

`GetSourcePtrByIndex` will default to use source at certain index or 1st source in Sources or spec.Source. The behavior is similar to `GetSource` function. Ref: https://github.com/argoproj/argo-cd/pull/17808/files#diff-1a32760cee151b5181abcd273fb66e286a02fd8c99573d988e423528760cc293R238-R245